internal/cli/atlas/search/nodes/delete.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodes
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type DeleteOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	*cli.DeleteOpts
+	store store.SearchNodesDeleter
+}
+
+func (opts *DeleteOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var deleteTemplate = "Started deleting search nodes for cluster '{{.}}'.\n"
+var deleteWatchTemplate = "Search nodes for cluster '{{.}}' are deleted.\n"
+
+func (opts *DeleteOpts) Run() error {
+	if err := opts.Prompt(); err != nil {
+		return nil
+	}
+
+	if !opts.Confirm {
+		fmt.Println(opts.FailMessage())
+		return nil
+	}
+	if err := opts.store.DeleteSearchNodes(opts.ConfigProjectID(), opts.Entry); err != nil {
+		return err
+	}
+
+	if opts.EnableWatch {
+		if err := opts.Watch(opts.watcher); err != nil {
+			return err
+		}
+		opts.Template = deleteWatchTemplate
+	} else {
+		opts.Template = deleteTemplate
+	}
+
+	return opts.Print(opts.Entry)
+}
+
+func (opts *DeleteOpts) watcher() (bool, error) {
+	_, err := opts.store.SearchNodes(opts.ConfigProjectID(), opts.Entry)
+	// searchNodes returns an error when there are no search nodes available
+	if err == nil {
+		return false, nil
+	}
+
+	if strings.Contains(err.Error(), "ATLAS_FTS_DEPLOYMENT_DOES_NOT_EXIST") {
+		return true, nil
+	}
```

Comment:
nit: you can simplify this a bit more

```golang
	if _, err := opts.store.SearchNodes(opts.ConfigProjectID(), opts.Entry); err != nil {
		if strings.Contains(err.Error(), "ATLAS_FTS_DEPLOYMENT_DOES_NOT_EXIST") {
		return true, nil
	         }
	}

```

Reply:
jumping here, `strings.Contains(error.Error())` is an anti-pattern
please use `admin.AsError(err)` like
```
target, ok := atlasv2.AsError(err)
	if ok && target.GetErrorCode() == "ATLAS_FTS_DEPLOYMENT_DOES_NOT_EXIST" {
		...
	} 
```